doc/connectivity/bluetooth/bluetooth-ctlr-arch.rst

```diff
@@ -135,6 +135,117 @@ Execution Priorities
+
+- struct llcp_mem_pool
+   * Mem pool used to implement procedure contexts resource - instantiated in both a local and a remote version
+   * Used through ull_llcp.c::create_procedure()
+
+- Miscellaneous pdu gymnastics
+   * Encoding and decoding of control pdus done by ull_llcp_pdu.c::llcp_pdu_encode/decode_<PDU>()
+   * Miscellaneous pdu validation handled by ull_llcp.c::pdu_validate_<PDU>() via ull_llcp.c::pdu_is_valid()
+
+LLCP local and remote request/procedure state machines
+======================================================
+
+- ull_llcp_local.c
+   * State machine handling local initiated procedures
+   * Naming concept: lr _<...> => local request machine
+   * Local procedure queue handling
+   * Local run/rx/tx_ack switch
+
+- ull_llcp_remote.c
+   * Remote versions of the above
+   * Naming concept: rr_<...> => remote request machine
+   * Also handling of remote procedure initiation by llcp_rx_new()
+   * Miscellaneous procedure collision handling (in rr_st_idle())
+
+- ull_llcp_common/conn_upd/phy/enc/cc/chmu.c
+   * Individual procedure implementations (ull_llcp_common.c collects the simpler ones)
+   * Naming concept: lp_<...> => local initiated procedure, rp_<...> => remote initiated procedure
+   * Handling of procedure flow from init (possibly through instant) to completion and host notification if applicable
+
+Miscellaneous concepts
+======================
+
+- Procedure collision handling
+   * See BT spec for explanation
+   * Basically some procedures can exist in parallel but some can't - for instance only one instant based at a time
+   * Spec states rules for how to handle/resolve collisions when they happen
+
+- Termination handling
+   * Specific rules apply re. how termination is handled.
+   * Since we have resource handling re. procedure contexts and terminate must always be available this is handled as a special case
+   * Note also - there are miscellaneous cases where connection termination is triggered on invalid peer behavior
+
+- New remote procedure handling
+   * Table new_proc_lut[] maps LLCP PDUs to procedures/roles used in llcp_rr_new()
+   * Note - for any given connection, there can only ever be ONE remote procedure in the remote procedure queue
+
+- Miscellaneous minors
+   * pause/resume concepts - there are two (see spec for details)
+   * procedure execution can be paused by the encryption procedure
+   * data TX can be paused by PHY, DLE and ENC procedure
+   * RX node retention - ensures no waiting for allocation of RX node when needed for notification
+
+
+Miscellaneous unit test concepts
+================================
+
+- Individual ZTEST unit test for each procedure
+   * zephyr/tests/bluetooth/controller/ctrl_<proc>
+
+- Only PDU flow is tested (as a main rule)
+   * Helpers emulate prepare/tx_ack/rx by event_prepare(), event_tx_ack() and event_done()
+   * PDU flow to/from controller is emulated by lt_rx/lt_tx and ut_rx interfacing with test rx/tx queues
```

Comment:
Isn't this just a "compacted" version of the details in L241-L245?

Reply:
Agree - removed 233-235